tools/generators/xcschemes/src/Generator/Generator.swift

```diff
@@ -54,7 +55,24 @@ struct Generator {
             targetsByKey: targetsByKey
         )
 
-        let schemeInfos = customSchemeInfos + automaticSchemeInfos
+        // Apply scheme auto-generation exclude patterns
+        let schemeInfos = (customSchemeInfos + automaticSchemeInfos).filter { scheme in
+            // Scheme name exclude patterns
+            let schemeName = scheme.name
+            let schemeNameFullRange = NSRange(schemeName.startIndex..., in: schemeName)
+
+            for pattern in arguments.autogenerationSchemeNameExcludePatterns {
+                let matches = try? NSRegularExpression(
+                    pattern: pattern
+                ).matches(in: schemeName, range: schemeNameFullRange)
```

Comment:
Can ew use the new Swift regex APIs?

Reply:
https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/3027/commits/42ebb66a067515ee6cfe2901842ac5f5d8026c80